engine.py

```diff
@@ -485,3 +484,146 @@ def _set_project(self):
+        
+        Additional parameters specified will be passed through to the widget_class constructor.
+        """
+        from tank.platform.qt import QtCore, QtGui
+        
+        tk_maya = self.import_module("tk_maya")
+        
+        panel_id = self._generate_panel_id(title, bundle)
+        widget_id = "ui_%s" % panel_id
+        self.log_debug("Begin showing panel %s" % panel_id)
+                                    
+        # create a maya window and layout
+        window = pm.window()
+        self.log_debug("Created window: %s" % window)
+        maya_layout = pm.formLayout(parent=window)
+        self.log_debug("Created layout %s" % maya_layout)
+        
+        if pm.control(widget_id, query=1, exists=1):
+            self.log_debug("Toolkit widget already exists. Reparenting it...")
+            # find the widget for later use
+            for widget in QtGui.QApplication.allWidgets():
+             if widget.objectName() == widget_id:
+                 widget_instance = widget
+            
+        else:
+            self.log_debug("Toolkit widget does not exist - creating it...")
+            parent = self._get_dialog_parent()            
+            widget_instance = widget_class(*args, **kwargs)
+            widget_instance.setParent(parent)
+            widget_instance.setObjectName(widget_id)
+            self.log_debug("Created %s (Object Name '%s')" % (widget_instance, widget_id))
+            # apply external stylesheet
+            self._apply_external_styleshet(bundle, widget_instance)
+                      
+        
+        # now reparent the widget instance to the layout
+        # we can now refer to the QT widget via the widget name
+        self.log_debug("Parenting widget %s to temporary window %s..." % (widget_id, maya_layout))
+        pm.control(widget_id, edit=True, parent=maya_layout)
+        
+        # now attach our widget in all four corners to the maya layout so that it fills 
+        # the entire panel space
+        pm.formLayout(maya_layout, 
+                      edit=True, 
+                      attachForm=[(widget_id, 'top', 1), 
+                                  (widget_id, 'left', 1), 
+                                  (widget_id, 'bottom', 1), 
+                                  (widget_id, 'right', 1)] )
+        
+        if pm.control(panel_id, query=1, exists=1):
+            # exists already - delete it
+            self.log_debug("Panel exists. Deleting it.")
+            pm.deleteUI(panel_id)
+                    
+        # lastly, ditch the maya window and host the layout inside a dock
+        
+        # see if the app has a preference where it likes to be docked:
+        if hasattr(widget_instance, "docking_preference"):
+            docking_area = widget_instance.docking_preference
+            self.log_debug("App docking preference found: %s" % docking_area)
+        else:
+            docking_area = "right"
```

Comment:
following engineering discussion, we'll leave it even more fixed for the time being and will look into adding a class to handle settings.